lib/views/question/question_view.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:kayla_flutter_ic/gen/assets.gen.dart';
+import 'package:kayla_flutter_ic/utils/app_bar_ext.dart';
+import 'package:kayla_flutter_ic/utils/route_paths.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_ui_model.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_view.dart';
+
+class QuestionView extends StatelessWidget {
+  final QuestionContainerUiModel uiModel;
+  final Widget child;
+  final Function() onNextQuestion;
+  final Function() onSubmit;
+
+  const QuestionView({
+    super.key,
+    required this.uiModel,
+    required this.child,
+    required this.onNextQuestion,
+    required this.onSubmit,
+  });
+
+  AppBar _appBar(BuildContext context) => AppBarExt.appBarWithCloseButton(
+        context: context,
+        onPressed: () => context.goNamed(
+          RoutePath.home.name,
```

Comment:
@nmint8m Should we use `pushReplacement`? `goNamed` does not close the current screen 🤔

Reply:
This is the result of `goName` and `pushReplacementName`. I believe that `goNamed` is better approach.

| `goName` |  `pushReplacementName` |
|----|----|
| <img width=300 src="https://user-images.githubusercontent.com/23162627/222390464-5b929180-d995-4daa-bc33-844da353095c.gif"> | <img width=300 src="https://user-images.githubusercontent.com/23162627/222391218-da60803a-9327-4d01-a00e-003e4fb58901.gif"> |

Reply:
@doannimble Oh, you're right, when I tap on the physical back button, it pops to the previous question.

I changed to `pushReplacementNamed` just like your suggestion. 👍